src/Spectre.libGenetic/GenerationFactory.cpp

```diff
@@ -0,0 +1,65 @@
+﻿/*
+* GenerationFactory.cpp
+* Builds different versions of Generation.
+*
+Copyright 2017 Grzegorz Mrukwa, Wojciech Wilgierz
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include "GenerationFactory.h"
+#include "Spectre.libException/ArgumentOutOfRangeException.h"
+#include <memory>
+#include <algorithm>
+
+namespace spectre::algorithm::genetic
+{
+GenerationFactory::GenerationFactory(size_t size, size_t individualSize, size_t initialFillup)
+    : m_Size(size),
+      m_IndividualSize(individualSize),
+      m_InitialFillup(initialFillup)
+{
+    if (initialFillup > individualSize)
+    {
+        throw core::exception::ArgumentOutOfRangeException<size_t>("initialFillup", 0, individualSize, initialFillup);
+    }
+}
+
+Generation GenerationFactory::buildDefaultGeneration(Seed seed) const
+{
+    std::vector<Individual> individuals;
+    for (auto i = 0u; i < m_Size; i++)
+    {
+        individuals.push_back(buildDefaultIndividual(seed + i));
```

Comment:
I'd go for generating seed step basing on the initial seed.

**What's the issue?**

When we'll try to make random `Generation`s with different `seed` in loop. Most likely similarly like you're doing here - it's the simplest way: `factory.buildDefaultGeneration(0)` and then `factory.buildDefaultGeneration(1)`.

In current implementation, there is only one `Individual` different in such generations, while even for small seed difference we should get absolutely different ones!

I'd propose something like this, or similar:
```
RNG rng(seed);
const auto seedIncrement = rng.next();
...
for(...)
{
    buildIndividual(seed + i * seedIncrement);
}
...
```

I hope, you grasp the idea.